subprojects/normalization-java/src/test/groovy/org/gradle/internal/normalization/java/ApiClassExtractorTestSupport.groovy

```diff
@@ -63,33 +64,33 @@ class ApiClassExtractorTestSupport extends Specification {
     }
 
     @CompileStatic
-    public static class ApiContainer {
+    static class ApiContainer {
         private final ApiClassLoader apiClassLoader = new ApiClassLoader()
         private final ApiClassExtractor apiClassExtractor
 
         public final Map<String, GeneratedClass> classes
 
-        public ApiContainer(List<String> packages, Map<String, GeneratedClass> classes) {
+        ApiContainer(List<String> packages, Map<String, GeneratedClass> classes) {
             this.apiClassExtractor = new ApiClassExtractor(packages.toSet())
             this.classes = classes
         }
 
         protected Class<?> extractAndLoadApiClassFrom(GeneratedClass clazz) {
-            apiClassLoader.loadClassFromBytes(apiClassExtractor.extractApiClassFrom(new ClassReader(clazz.bytes)))
+            apiClassLoader.loadClassFromBytes(apiClassExtractor.extractApiClassFrom(new ClassReader(clazz.bytes)).get())
         }
 
         protected byte[] extractApiClassFrom(GeneratedClass clazz) {
-            apiClassExtractor.extractApiClassFrom(new ClassReader(clazz.bytes))
+            apiClassExtractor.extractApiClassFrom(new ClassReader(clazz.bytes)).get()
         }
 
         protected boolean shouldExtractApiClassFrom(GeneratedClass clazz) {
```

Comment:
I think most of this assertions are now superfluous, since before checking the `should` method, the extract method is called which would fail if the result is not present. I think we can get rid of the method altogether.

Reply:
Well spotted. I removed the ones that made no sense anymore, I kept a few that still do.